Source/WebKit/WebProcess/WebProcess.cpp

```diff
@@ -614,6 +614,12 @@ void WebProcess::initializeWebProcess(WebProcessCreationParameters&& parameters,
         RetainPtr<CFDataRef> auditData = adoptCF(CFDataCreate(nullptr, (const UInt8*)&*auditToken, sizeof(*auditToken)));
         Inspector::RemoteInspector::singleton().setParentProcessInformation(WebCore::presentingApplicationPID(), auditData);
     }
+    // We need to connect to webinspectord before the first page load for the XPC connection to be successfully opened.
+    // This is because we block launchd before the first page load, and launchd is required to establish the connection.
+    // This is only done if Web Inspector is enabled, which is determined by the size of the Web Inspector extension vector.
+    // See WebProcessProxy::shouldEnableRemoteInspector().
+    if (parameters.enableRemoteWebInspectorExtensionHandles.size())
```

Comment:
is it possible for `parameters.enableRemoteWebInspectorExtensionHandles` to be consumed before this is reached?  i see some code in `WebProcess:: platformInitializeWebProcess` that seems like it will affect this, but not entirely sure if `.size()` would still be valid even after that

Reply:
That's a good point! The sandbox extensions are consumed before this point, but the vector is left unchanged, so `size()` is still be valid.